### PR TITLE
[DOC] Doc for command line options

### PR DESCRIPTION
**Pull request by BurdetteLamar**
Doc for certain Ruby command-line options that affect the program's environment.

### CODE REVIEW

File: doc/command_line/environment.md [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+the option may be given more than once:
+
+```sh
+$ pushd C:/
+$ ruby -e 'p $LOAD_PATH.size'
+8
+$ $ ruby -I my_lib -I some_lib -e 'p $LOAD_PATH.size; p $LOAD_PATH[0..1]'
+10
+["C:/my_lib", "C:/some_lib"]
+$ popd
```
Comment (peterzhu2118) [inline doc/command_line/environment.md:45]: Can we have this example on a file system that's not Windows? Most developers are using a UNIX system and not a Windows one so the path may look odd.

File: doc/command_line/environment.md [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+
+The argument to option `-I` specifies a directory
+to be added to the array in global variable `$LOAD_PATH`;
+the option may be given more than once:
+
+```sh
+$ pushd C:/
+$ ruby -e 'p $LOAD_PATH.size'
+8
+$ $ ruby -I my_lib -I some_lib -e 'p $LOAD_PATH.size; p $LOAD_PATH[0..1]'
```
Comment (peterzhu2118) [inline doc/command_line/environment.md:42]:
```suggestion
$ ruby -I my_lib -I some_lib -e 'p $LOAD_PATH.size; p $LOAD_PATH[0..1]'
```

Comment (peterzhu2118) [inline doc/command_line/environment.md:42]: You can use `Array#take`

```suggestion
$ $ ruby -I my_lib -I some_lib -e 'p $LOAD_PATH.size; p $LOAD_PATH.take(2)'
```

File: doc/command_line/environment.md [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+```sh
+$ ruby -e 'p $DEBUG'
+false
+$ ruby -d -e 'p $DEBUG'
+true
+```
+
+### Option '-w'
+
+Option `-w` (lowercase letter) is equivalentto option `-W1` (uppercase letter).
```
Comment (peterzhu2118) [inline doc/command_line/environment.md:117]:
```suggestion
Option `-w` (lowercase letter) is equivalent to option `-W1` (uppercase letter).
```

File: doc/command_line/environment.md [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+
+### Option `-W`
+
+Ruby code can create a <i>warning message</i> by calling method Kernel#warn;
+this may cause a message to be printed on `$stderr`
+(or not, depending on certain settings).
```
Comment (peterzhu2118) [inline doc/command_line/environment.md:123]: This also turns on and off Ruby's warning. For example, on Ruby 3.3:

```shell
$ RUBY_FREE_AT_EXIT=1 ruby -e ""
ruby: warning: Free at exit is experimental and may be unstable
$ RUBY_FREE_AT_EXIT=1 ruby -W0 -e ""
```

File: doc/command_line/environment.md [hunk 5]
```diff
@@ -0,0 +1,181 @@
...
+
+Option `-W` helps determine whether a particular warning message
+will be written,
+by setting the initial value of environment variable `$-W`:
```
Comment (peterzhu2118) [inline doc/command_line/environment.md:127]: You mean global variable right?

```suggestion
by setting the initial value of global variable `$-W`:
```

File: doc/command_line/environment.md [hunk 6]
```diff
@@ -0,0 +1,181 @@
...
+#<IO::Buffer>
+$ ruby -W1 -e 'p $-W; p IO::Buffer.new'
+1
+-e:1: warning: IO::Buffer is experimental and both the Ruby and C interface may change in the future!
+#<IO::Buffer>
+$ ruby -W2 -e 'p $-W; p IO::Buffer.new'
+2
+-e:1: warning: IO::Buffer is experimental and both the Ruby and C interface may change in the future!
+#<IO::Buffer>
+```
```
Comment (peterzhu2118) [inline doc/command_line/environment.md:150]: `IO::Buffer` may become non-experimental in the future, so let's not use it as an example.

We can use `Kernel#warn` for some examples. Another example is:

```sh
$ ruby -W1 -e 'p $foo'
nil
$ ruby -W2 -e 'p $foo'
-e:1: warning: global variable `$foo' not initialized
nil
```

File: doc/command_line/environment.md [hunk 7]
```diff
@@ -0,0 +1,181 @@
...
+
+You can suppress a category by prefixing `no-` to the category name:
+
+```
+$ ruby -W:no-experimental -e 'p IO::Buffer.new'
+#<IO::Buffer>
+```
+
+See also {Field Processing}[rdoc-ref:command_line/field_processing.md].
```
Comment (peterzhu2118) [inline doc/command_line/environment.md:181]: I'm confused as to why this is here. How does "Field Processing" relate to this?

File: doc/command_line/environment.md (hunk 1 above)
Comment (BurdetteLamar) [reply doc/command_line/environment.md]: Fixed.

File: doc/command_line/environment.md (hunk 2 above)
Comment (BurdetteLamar) [reply doc/command_line/environment.md]: Fixed.

File: doc/command_line/environment.md (hunk 3 above)
Comment (BurdetteLamar) [reply doc/command_line/environment.md]: Fixed.

File: doc/command_line/environment.md (hunk 4 above)
Comment (BurdetteLamar) [reply doc/command_line/environment.md]: Revised.

File: doc/command_line/environment.md (hunk 5 above)
Comment (BurdetteLamar) [reply doc/command_line/environment.md]: Right.  FIxed.

File: doc/command_line/environment.md (hunk 6 above)
Comment (BurdetteLamar) [reply doc/command_line/environment.md]: Example replaced.

File: doc/command_line/environment.md (hunk 7 above)
Comment (BurdetteLamar) [reply doc/command_line/environment.md]: Removed.

File: doc/command_line/environment.md [hunk 8]
```diff
@@ -0,0 +1,170 @@
...
+### Option `-0`
+
+Option `-0` defines the input record separator `$/`
+for the invoked Ruby program.
+
+The optional argument to the option must be octal digits,
+each in the range `0..7`;
+these digits are prefixed with digit `0` to form an octal value:
+
+- If that value is in range `(0..0377)`,
```
Comment (nobu) [inline doc/command_line/environment.md:77]:
```suggestion
- If that value is `0`, the input record separator is `""`;
  see {Special Line Separator Values}[rdoc-ref:IO@Special+Line+Separator+Values].
- If that value is in range `(1..0377)`,
```

File: doc/command_line/environment.md [hunk 9]
```diff
@@ -0,0 +1,170 @@
...
+- Option not given: Same as `-W1` (moderate verbosity).
+
+The value of `$-W`, in turn, determines which warning messages (if any)
+are to be printed to `$stdout` (see Kernel#warn):
+
+```sh
+$ ruby -W1 -e 'p $foo'
+nil
+$ ruby -W2 -e 'p $foo'
+-e:1: warning: global variable `$foo' not initialized
```
Comment (nobu) [inline doc/command_line/environment.md:144]: The quote has changed recently.

```suggestion
-e:1: warning: global variable '$foo' not initialized
```

File: doc/command_line/environment.md (hunk 8 above)
Comment (BurdetteLamar) [reply doc/command_line/environment.md]: Revised.

File: doc/command_line/environment.md (hunk 9 above)
Comment (BurdetteLamar) [reply doc/command_line/environment.md]: Fixedd.

**Review by peterzhu2118 — APPROVED** [review]

**Pull request merged**